### ISSUE TITLE
CREATE_FAILED with "Domain cannot contain reserved word: aws"

### ISSUE THREAD

**Issue by hakanson** [aws-solutions/content-analysis-on-aws#8]
tl;dr - don't let your stack name contain "aws"

---

The `MieCognitoDomain` step had a with `"Domain cannot contain reserved word: aws (Service: AWSCognitoIdentityProviderService; Status Code: 400; Error Code: InvalidParameterException; Request ID: [redacted]; Proxy: null)"`

I found the error in CloudTrail CreateUserPoolDomain event:
```json
    "eventName": "CreateUserPoolDomain",
    "errorCode": "InvalidParameterException",
    "errorMessage": "Domain cannot contain reserved word: aws",
    "requestParameters": {
        "userPoolId": "us-east-1_HYsU8EgOr",
        "domain": "aws-content-analysis-dataplane-1hocmg2w2zumk"
    },
```

And I realized when naming my stack, I choose "aws-content-analysis" and `AWS::StackName` was implicitly used at some point.

I'm not sure if any check can be done, but writing this issue in case anyone else hits this problem and searches for the error message.
